src/DataHolder/CurrencyHolder.php

```diff
@@ -60,20 +65,38 @@ public function getPrecision(string $currency): int
     }
 
     /**
-     * @inheritDoc
+     * @param float $amount
+     * @param string $currencyCode
+     * @return float
+     * @throws NoCurrencyRateException
      */
     public function exchangeToBase(float $amount, string $currencyCode): float
     {
         $rate = $this->getRate($currencyCode);
-        return (float) bcdiv((string) $amount, (string) $rate, 4);
+        if ($rate === null) {
```

Comment:
This and `exchangeFromBase` seems to be the same. Maybe creating a function would help.